Source/objects.cpp

```diff
@@ -509,7 +509,7 @@ void AddBookLever(int x1, int y1, int x2, int y2, _speech_id msg)
 		AddObject(OBJ_BLOODBOOK, { xp, yp });
 	}
 	int ob = dObject[xp][yp] - 1;
-	SetObjMapRange(ob, x1, y1, x2, y2, leverid);
+	Objects[ob].SetMapRange({ x1, y1 }, { x2, y2 }, leverid);

```

Comment:
Have you considered making `SetMapRange` only take a `Rectangle`, and then creating a `Rectangle` ctor that takes the topleft and bottomright points to build it?

I think that would centralize more of the logic in the rectangle class and this could be used in other places as well, while keeping consumer APIs simpler (no need for overloads in this case).

Reply:
I was a bit hesitant about adding that constructor in this PR as it would make declarations/initialisations like `Rectangle{ { 1, 2 }, { 4, 4 } }` ambiguous, forcing the use of Size or Point explicitly in those cases. It also makes the SetMapRange logic more verbose unless we also updated the save/load code to detect when these variables are used as map coordinates as we need to translate from Point/Size to Point/Point in at least one of those places. It would make it nicer to use in consumers (such as ObjChangeMap/Resync, Obj*Special, Add*Objs) though, so I reckon it would be worthwhile eventually.